tests/s3/copy_object/test_copy_object_quota.py

```diff
@@ -0,0 +1,724 @@
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+"""S3 copy object quota test module."""
+
+import logging
+import os
+from http import HTTPStatus
+
+import pytest
+
+from commons import configmanager
+from commons import error_messages as errmsg
+from commons.exceptions import CTException
+from commons.params import TEST_DATA_FOLDER
+from commons.utils import assert_utils
+from commons.utils.system_utils import path_exists, make_dirs
+from config.s3 import S3_CFG
+from libs.csm.csm_interface import csm_api_factory
+from libs.csm.csm_setup import CSMConfigsCheck
+from libs.ha.ha_common_libs_k8s import HAK8s
+from libs.s3 import s3_misc
+from libs.s3.s3_multipart_test_lib import S3MultipartTestLib
+
+LOGGER = logging.getLogger(__name__)
+
+
+# pylint: disable=too-many-public-methods
+class TestCopyObjectsQuota:
+    """S3 copy object class."""
+    # pylint: disable=too-many-arguments
+    # pylint: disable-msg=too-many-statements
+    # pylint: disable=too-many-arguments
+    # pylint: disable-msg=too-many-locals
+    @classmethod
+    def setup_class(cls):
+        """
+        Setup_class will be invoked prior to each test case.
+
+        It will perform all prerequisite test suite steps if any.
+        """
+        cls.log = logging.getLogger(__name__)
+        cls.log.info("STARTED: Class setup.")
+        cls.csm_obj = csm_api_factory("rest")
+        cls.ha_obj = HAK8s()
+        cls.config = CSMConfigsCheck()
+        cls.log.info("Initiating Rest Client ...")
+        cls.csm_conf = configmanager.get_config_wrapper(fpath="config/csm/test_rest_capacity.yaml")
+        cls.rest_resp_conf = configmanager.get_config_wrapper(
+            fpath="config/csm/rest_response_data.yaml")
+        cls.created_iam_users = set()
+        cls.buckets_created = []
+        cls.akey = None
+        cls.skey = None
+        cls.src_bkt = None
+        cls.dest_bkt1 = None
+        cls.obj = None
+        cls.user_id = None
+        setup_ready = cls.config.check_predefined_csm_user_present()
+        if not setup_ready:
+            setup_ready = cls.config.setup_csm_users()
+        assert setup_ready
+        resp = cls.ha_obj.get_config_value(cls.csm_obj.master)
+        if resp[0]:
+            cls.nvalue = int(resp[1]['cluster']['storage_set'][0]['durability']['sns']['data'])
+        cls.aligned_size = 4 * cls.nvalue
+        cls.test_file = "mp_obj"
```

Comment:
Let's add a timestamp to file and any other resource names in the module - 
```suggestion
        cls.test_file = f"mp_obj_{time.perf_counter_ns()}"
```